### PR TITLE
IDTCTJNKNS-201: Initial add of tests for DetectJarManager

### PR DESCRIPTION
**Pull request by DMaxfield-BDS**

### CODE REVIEW

File: src/test/java/com/synopsys/integration/jenkins/detect/substeps/DetectJarManagerTest.java [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+        Mockito.when(taskListener.getLogger()).thenReturn(new PrintStream(byteArrayOutputStream));
+        JenkinsIntLogger jenkinsIntLogger = new JenkinsIntLogger(taskListener);
+
+        if (logLevel != null) {
+            jenkinsIntLogger.setLogLevel(logLevel);
+        }
+
+        try {
+            // Perform same actions as calculateJavaExecutablePath().
+            String adjustedJavaHome = javaHomeInput == null ? "java" : new File(javaHomeInput + "/bin/java").getCanonicalPath();
```
Comment (rottebds) [inline src/test/java/com/synopsys/integration/jenkins/detect/substeps/DetectJarManagerTest.java:75]: Replicating the functionality of calculateJavaExecutablePath() is not the best way to test that it does what we expect. We should instead pass in an "expectedCalculatedJavaHome" or something to that effect that we can validate against the input.

File: src/test/java/com/synopsys/integration/jenkins/detect/substeps/DetectJarManagerTest.java [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+            Arguments.of(testJavaHome, false, testDetectJarPath, testPathEnv, LogLevel.TRACE),
+            Arguments.of("", false, "", "", LogLevel.DEBUG),
+            Arguments.of(" ", false, " ", " ", LogLevel.DEBUG),
+            Arguments.of("", false, "", "", null)
+        );
+    }
+
+    @ParameterizedTest
+    @MethodSource({ "setUpForExecutionInput" })
+    public void testSetUpForExecution(String javaHomeInput, Boolean isValidJavaHome, String detectJarPathInput, String pathEnvInput, LogLevel logLevel) {
```
Comment (rottebds) [inline src/test/java/com/synopsys/integration/jenkins/detect/substeps/DetectJarManagerTest.java:61]: I'm not sure if this test should be a parameterized test. The test is largely lines 77 and 78, with a variable number of assertions on the lines following. Lines 62 through 75 all seem to be setup. Is this implementation preferable to a common method that performs the test and several different 6-7 line tests that are very clear in what they assert?

File: src/test/java/com/synopsys/integration/jenkins/detect/substeps/DetectJarManagerTest.java [hunk 3]
```diff
@@ -26,82 +21,115 @@
...
-
-        if (logLevel != null) {
-            jenkinsIntLogger.setLogLevel(logLevel);
-        }
-
-        try {
-            // Perform same actions as calculateJavaExecutablePath().
-            String adjustedJavaHome = javaHomeInput == null ? "java" : new File(javaHomeInput + "/bin/java").getCanonicalPath();
+    @Test
+    public void testSetUpForExecutionNullInput() {
```
Comment (rottebds) [inline src/test/java/com/synopsys/integration/jenkins/detect/substeps/DetectJarManagerTest.java:61]: At first glance, this test and the two tests preceding it are the same test, but with a different java home / jar path. However, looking further into it, you're also testing different Java paths and asserting different results. Unless there's a compelling reason not to, it seems likely that we should combine these three tests into a parameterized test that tests " ", "", and null across the same other lines of test code.

File: src/test/java/com/synopsys/integration/jenkins/detect/substeps/DetectJarManagerTest.java [hunk 4]
```diff
@@ -26,82 +21,115 @@
...
+    private final JenkinsIntLogger jenkinsIntLogger = new JenkinsIntLogger(taskListener);
 
     private final ByteArrayOutputStream byteArrayOutputStream = new ByteArrayOutputStream();
-    private final PrintStream originalOut = System.out;
 
     @BeforeEach
     public void setUp() {
-        System.setOut(new PrintStream(byteArrayOutputStream));
+        environmentVariables.put("PATH", pathEnvVarInput);
+        Mockito.when(taskListener.getLogger()).thenReturn(new PrintStream(byteArrayOutputStream));
```
Comment (rottebds) [inline src/test/java/com/synopsys/integration/jenkins/detect/substeps/DetectJarManagerTest.java:41]: Is there any particular reason that we're doing this work here as opposed to in createAndTestDetectSetupResponse?

File: src/test/java/com/synopsys/integration/jenkins/detect/substeps/DetectJarManagerTest.java (hunk 3 above)
Comment (DMaxfield-BDS) [reply src/test/java/com/synopsys/integration/jenkins/detect/substeps/DetectJarManagerTest.java:61]: The 3rd test (testSetUpForExecutionNullInput) calls this.**valid**JavaExecValidation(), the other 2 call this.**invalid**JavaExecValidation(). I didn't think the added complexity to handle both cases OR a parameterized test for 2 tests was worth it here.

Comment (rottebds) [reply src/test/java/com/synopsys/integration/jenkins/detect/substeps/DetectJarManagerTest.java:54]: Ah, likely a good call. I missed the distinction in validation.

**Pull request merged**